src/nvim/tui/tui.c

```diff
@@ -595,12 +611,31 @@ static void print_cell(UI *ui, UCell *ptr)
     // Printing the next character finally advances the cursor.
     final_column_wrap(ui);
   }
-  update_attrs(ui, kv_A(data->attrs, ptr->attr));
-  out(ui, ptr->data, strlen(ptr->data));
-  grid->col++;
-  if (data->immediate_wrap_after_last_column) {
-    // Printing at the right margin immediately advances the cursor.
-    final_column_wrap(ui);
+  // If the terminal wrapped in the last column doesn't have scroll forward,
+  // the last cell can not be outputed.
+  // FIXME If BCE is enabled, the display may be distorted if you set
+  // termguicolors. Also, if BCE can not be used, the last cell is not output,
+  // so if you set termguicolors, the last cell will not have color.
+  bool lastcell = grid->row == ui->height - 1 && grid->col == ui->width - 1;
```

Comment:
nvim core will never draw something in the last cell, this should only come from an emulated clear. It might be worth waiting with this fix until after #9193, where this case is separated out in `clear_region`